### ISSUE TITLE
Question docked at bottom #2

### ISSUE THREAD

**Issue by linkjay** [enquirer/enquirer#37]
Now that it is released, I must ask again:

How exactly can I make a prompt that is sort of docked at the bottom? Mainly just want any of my output to go above the prompt.

**Comment by jonschlinkert** [comment]
Can you point to an example of something similar? Even if it's not a prompt, just so I can visualize what you mean. Once I can see it in my mind, I can create it.

**Comment by jonschlinkert** [comment]
I'm still struggling to visualize what you mean. It sounds like you want to dock a message at the bottom of the terminal for the duration of multiple prompts, not just during a single prompt. Is that correct?

**Comment by linkjay** [comment]
Either. Generally like if you output anything on a prompt, it will not work correctly. The text will go into the prompt and under it and you can't type in it correctly after that.

I want to be able to take any sort of readline prompt and have any outputted text not interfere with the prompt.

Heres an example with enquirer 2.0.7.

https://i.linkjay.io/wbjhkD.png

As you can see, the prompt will move it the whole thing to a new line and it gets in the way and looks very awkward. Maybe there is a way you could jump on an stdout event or something to move any sort of console.log above the prompt instead of recreating it or whatever is happening. Ideally, it would be nice if the prompt could look like this:

OUTPUT TEXT
OUTPUT TEXT 
OUTPUT TEXT 

? > can type text here and all OUTPUT TEXT will go above this prompt

**Comment by jonschlinkert** [comment]
Sorry that it took me so long to respond. The behavior you're seeing is correct, since you're trying to use `console.log()` with the prompt. Instead, try using the `footer` option. 

See [this example](https://github.com/enquirer/enquirer/blob/69deef48c8d7535c5175f8b1b981638830541821/examples/input/option-header-footer-initial.js), and please let me know if that's what you meant. Thanks!

**Comment by linkjay** [comment]
No, I'm sorry I must be more than confusing here.

I understand that this is desired behavior, and that example does not show my issue. In that example, that shows output being drawn under the prompt, but I am not interested in that. I am asking if there is a way that upon console.log or any stdout, the prompt is moved down so the stdout is actually drawn above that prompt.

Think about having a terminal console. The typing input is always on the bottom, and any stdout that comes will be pasted above that.

**Comment by JcBernack** [comment]
@jonschlinkert Imagine an async process trying to log anything to the console while a prompt is open:

```
const prompt = new Select({
    name: 'color',
    message: 'Pick a flavor',
    choices: ['apple', 'grape', 'watermelon', 'cherry', 'orange']
});

setInterval(() => console.log('FOO'), 1000);

prompt.run()
    .then(answer => console.log('Answer:', answer))
    .catch(console.error);
```

The `FOO` message will be written over the open prompt at the bottom. If you move the cursor within the prompt it will refresh and overwrite the `FOO`s. Inquirer has exactly the same issue btw.

**Comment by linkjay** [comment]
@JcBernack Yes, exactly! Thank you for re-explaining. I hope this clears things up @jonschlinkert
